docs/authoring-recipes/recipe-development-environment.md

```diff
@@ -283,6 +279,43 @@ tasks.compileJava {
 </TabItem>
 </Tabs>
 
+### Jackson deserialization
+
+Some recipe implementations may use [Jackson deserialization](https://github.com/FasterXML/jackson-annotations) by defining annotations such as `@JsonCreator` and `@JsonProperty`.
+
+In order to deserialize Java classes, you need to add the `-parameters` compiler argument. This is necessary for Jackson to be able to deserialize the constructor parameters of a class.
+
+<Tabs groupId="projectType">
+<TabItem value="gradle" label="Gradle">
+
+```groovy title="build.gradle"
+tasks.withType(JavaCompile).configureEach {
+    options.compilerArgs.add('-parameters')
+}
+```
+</TabItem>
+
+<TabItem value="maven" label="Maven">
+
+```xml title="pom.xml"
+<build>
+    <plugins>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.13.0</version>
```

Comment:
There's been [a newer release](https://maven.apache.org/plugins/maven-compiler-plugin/). Are we ok to adopt that? [System requirements](https://maven.apache.org/plugins/maven-compiler-plugin/plugin-info.html#system-requirements-history) seems to think so.
```suggestion
            <version>3.14.0</version>
```
I'd considered leaving out a version, but then you likely get a much older parent pom managed version; not great.